modules/local/transrate.nf

```diff
@@ -0,0 +1,42 @@
+process TRANSRATE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::transrate=1.0.3"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/transrate:1.0.3--hec16e2b_4':
+        'biocontainers/transrate:1.0.3--hec16e2b_4' }"
+
+    input:
+    tuple val(meta), path(assembly)
+
+    output:
+    tuple val(meta), path("*assemblies_mqc.csv") , emit: assembly_qc
+    path "versions.yml"                          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    // transrate flashes a warning about a ruby gem being out of date, so call the version before it is being piped into the yaml
+    """
+    gzip -d -c $assembly > assembly_unzipped.fa
+
+    transrate \\
+        --threads $task.cpus \\
+        --assembly assembly_unzipped.fa \\
+        --output ${prefix}_transrate \\
+        $args
```

Comment:
A few thoughts:
1. nf-core modules should take gzipped files, so from that point of view unzipping in the module is fine. I think this would actually be a great addition to nf-core modules.
2. For other purposes, one could call the `UNPIGZ` module before instead. I know that they preferred that in the mag pipeline because more than one module wanted unzipped input and by calling `UNPIGZ` they were left with only one unzipped file instead of one in each work directory where it was needed. _Edit:_ Now I see that we actually call `UNPIGZ`. You should use the output channel from that as input to this and remove the `gunzip`.
3. You can't trick `transrate` to read a compressed file with `--assembly <(gunzip -c $assembly)`?

Reply:
I am a little confused here, you say the transrate module is good to take in gzipped files, but not in this case of the meta-t pipeline, since we're passing in UNPIGZ?

Reply:
Also, if we're using `UNPIGZ_CONTIGS.out.unzipped`, we should use the transrate module in the `CAT` block on line 455 yes?

Reply:
I have seen some other modules test for gzipped-ness and proceed accordingly, that might be the best general purpose move here.

Reply:
Sorry, I meant the first comment as a discussion start so it got a bit confusing. On top of that, I had missed that the `UNPIGZ_CONTIGS` was inside the `CAT` part, meaning it won't be available to `TRANSRATE`.

There are a couple of routes we could take: 1) Keep your code as it originally was, 2) move the call to `UNPIGZ_CONTIGS` to outside the `CAT` context and change `TRANSRATE` so it takes unzipped files. (I thought of another alternative, but can't remember now...)

I'm rather fond of number 2. `TRANSRATE` will always be called so unzipping will be needed at least once, so we're not unzipping unnecessarily. If the user decides to use `CAT`, we're saving one unzipping operation. There might be other modules that could be called with the output from `UNPIGZ_CONTIGS` (that take either zipped or unzipped). We could look for such cases later to save on unzipping.

If you decide to contribute the module to nf-core, you might want to rewrite it to take either zipped or unzipped. We can then keep the call as it is, but if we decide to let `CAT` go (which we might well do) we can drop `UNPIGZ_CONTIGS`. (In this case, I'm fond of the `<(gunzip -c file.gz)` syntax as it doesn't leave traces.)

Reply:
I agree, the second option is the right way to go! I implemented and it seems to run smoothly, take a look and see if I changed what you had in mind.